core/iwasm/common/wasm_memory.h

```diff
@@ -14,6 +14,11 @@
 extern "C" {
 #endif
 
+#define GET_LINEAR_MEMORY_SIZE(memory) \
+    BH_ATOMIC_32_LOAD(memory->memory_data_size)
+#define SET_LINEARM_MEMORY_SIZE(memory, size) \
+    BH_ATOMIC_32_STORE(memory->memory_data_size, size)
+
```

Comment:
The BH_ATOMIC_32_LOAD/STORE are only needed when shared memory is enabled, and might be not available in some platforms, how about:
```C
#if WASM_ENABLE_SHARED_MEMORY != 0
#define GET_LINEAR_MEMORY_SIZE(memory) \
    BH_ATOMIC_32_LOAD(memory->memory_data_size)
#define SET_LINEARM_MEMORY_SIZE(memory, size) \
    BH_ATOMIC_32_STORE(memory->memory_data_size, size)
#else
#define GET_LINEAR_MEMORY_SIZE(memory) memory->memory_data_size
#define SET_LINEARM_MEMORY_SIZE(memory, size) memory->memory_data_size = size
#endif